lib/linters.py

```diff
@@ -0,0 +1,627 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from collections import namedtuple, defaultdict
+from copy import deepcopy
+import datetime
+import json
+import re
+
+from lib import nacaddr
+import yaml
+
+Error = namedtuple('Error', ['filename', 'lineno', 'offset', 'Severity',
+                             'message', 'category'])
+
+
+def sanitizeNetworkItem(x):
+  # Remove comments and identify just the content we care about
+  return x.split('#', 1)[0].strip()
+
+
+class Severity(object):
+  """Helper class to represent the severity of a lint problem.
+
+  Methods:
+    human(k) - represents the human-legible value
+    int(k) - represents a numerical representation of how bad the error is,
+             mimicing the same values used for Phabricator.
+  """
+
+  DISABLED = 'disabled'
+  ADVICE = 'advice'
+  WARNING = 'warning'
+  AUTOFIX = 'autofix'
+  ERROR = 'error'
+
+  @classmethod
+  def human(cls, k):
+    mapping = {
+        cls.DISABLED: 'Disabled',
+        cls.ADVICE: 'Advice',
+        cls.WARNING: 'Warning',
+        cls.ERROR: 'Error',
+        cls.AUTOFIX: 'Auto-Fix',
+    }
+    return mapping[k]
+
+  @classmethod
+  def int(cls, k):
+    mapping = {
+        cls.DISABLED: 10,
+        cls.ADVICE: 20,
+        cls.WARNING: 30,
+        cls.ERROR: 40,
+        cls.AUTOFIX: 25,
+    }
+    return mapping[k]
+
+
+class LintErrors(object):
+  """Storage class for tracking what lint errors have been observed."""
+
+  def __init__(self):
+    self.errs = []
+    self.filename = None
+
+  def __iter__(self):
+    for e in self.errs:
+      yield e
+
+  def add(self, severity_code, msg, lineno=0, offset=0, filename=None,
+      category='CAP'):
+    """Construct a lint error based off the given location provided"""
+    fn = filename or self.filename or 'unknown'
+    self.errs.append(Error(fn, lineno, offset, severity_code, msg, category))
+
+  def tree(self, sort=True):
+    """Return a tree representation of the stored lint errors"""
+    t = defaultdict(list)
+    for e in self.errs:
+      t[e.filename].append(e)
+    if sort:
+      for v in t.values():
+        v.sort(key=lambda e: e.lineno)
+    return t
+
+  def pprint(self):
+    """Output a pretty-print of the stored lint errors"""
+    for filename, errors in self.tree().iteritems():
+      print("%s -" % filename)
+      for error in errors:
+        print("  %5d <%s> %s" % (error.lineno,
+                     Severity.human(error.Severity),
```

Comment:
This is aligned incorrectly. Aligning it correctly would look pretty bad so how about something more like

```
        print("  %5d <%s> %s" %
              (error.lineno, Severity.human(error.Severity), error.message))
```